operator/internal/manifests/storage/configure.go

```diff
@@ -187,68 +166,26 @@ func ensureObjectStoreCredentials(p *corev1.PodSpec, opts Options) corev1.PodSpe
 		}
 	case lokiv1.ObjectStorageSecretS3:
 		storeEnvVars = []corev1.EnvVar{
-			{
-				Name: EnvAWSAccessKeyID,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAWSAccessKeyID,
-					},
-				},
-			},
-			{
-				Name: EnvAWSAccessKeySecret,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAWSAccessKeySecret,
-					},
+			envVarFromSecret(EnvAWSAccessKeyID, secretName, KeyAWSAccessKeyID),
+			envVarFromSecret(EnvAWSAccessKeySecret, secretName, KeyAWSAccessKeySecret),
+		}
+		// STS Auth Workflow
+		if wifEnabled(opts) {
+			storeEnvVars = []corev1.EnvVar{
```

Comment:
I almost read that we added all four env vars when `wifEnabled` returns `true`. This whole switch case is hard to read. Need to :thinking: about possible improvements.